### PR TITLE
CP-35941: show newly added fields of records through xe 

### PR DESCRIPTION
**Pull request by psafont**
Also does some clean-ups and minor fixes:

- generate an empty list when setting ipvx_allowed is empty, the code actually depends on it being an empty list for some cases
- be consistent when showing lists of references, some of them weren't separated by spaces

### CODE REVIEW

**Comment by psafont** [comment]
There are still inconsistencies when lists are shown in the interface (in the get functions, is this something accidental or something that's wanted?) @robhoes

For example:
https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi-cli-server/records.ml#L955
https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi-cli-server/records.ml#L2267

**Review by lindig — APPROVED** [review]

File: ocaml/xapi-cli-server/records.ml [hunk 1]
```diff
@@ -162,6 +160,18 @@ let get_name_from_ref r =
...
 
+let get_uuid_from_ref_or_null r =
+  if r = Ref.null then
+    nullref
+  else
+    get_uuid_from_ref r
+
+let concat_to_string = String.concat "; "
+
+let get_string_from_list f rs = concat_to_string (List.map f rs)
```
[the remark this answers is not in the document]
Comment (psafont) [reply ocaml/xapi-cli-server/records.ml:171]: How about `map_and_concat`? This makes it easy to understand what the function argument does and how it collapses it to a single item.

[the remark this answers is not in the document]
Comment (lindig) [reply ocaml/xapi-cli-server/records.ml]: That's good, too.

**Review by lippirk — APPROVED** [review]
> xapi-cli-server: be consistent when showing lists 

I like this change; it doesn't seem to me that it will actually change the output out of xe, so hopefully it won't break anyone's scripts

The rest looks good

**Comment by psafont** [comment]
> it doesn't seem to me that it will actually change the output out of xe

It does change it, but being more consistent should make it easier to script.

**Review by robhoes — APPROVED** [review]

**Pull request merged**